apps/workers/db-check/worker.go

```diff
@@ -0,0 +1,157 @@
+func (w *worker) DoTask(ctx context.Context) (task.Result, string) {
+	records, err := w.RecordCount(ctx, w.DateField, w.Date1)
+	if err != nil {
+		return task.Failed(err)
+	}
+
+	if records == 0 {
+		m := []string{"no records found",
+			fmt.Sprintf("no `%s` for `%s`", w.DateField, w.Date1.Format("2006-01-02")),
+		}
+		w.sendSlack(m...)
+		return task.Completed(fmt.Sprintln(m))
+	}
+
+	var p float64
+	if w.Offset != "" {
+		w.Date2, err = date(w.Date1, w.Offset)
+		if err != nil {
+			return task.Failed(err)
+		}
+	}
+	var msg string
+	if !w.Date2.IsZero() {
+		records2, err := w.RecordCount(ctx, w.DateField, w.Date2)
+		if err != nil {
+			return task.Failed(err)
+		}
+
+		if records2 == 0 {
+			m := []string{"no comparison date records found",
+				fmt.Sprintf("no `%s` for `%s`", w.DateField, w.Date2.Format("2006-01-02")),
+			}
+			w.sendSlack(m...)
+			return task.Completed(fmt.Sprintln(m))
+		}
+
+		// if the change between compare and offset record counts is greather than the given percentage
+		// send an alert to slack that the variance is greater than expected
+		p = (float64(records2) - float64(records)) / float64(records)
+		if math.Abs(p) > w.Percent {
+			m := []string{
+				"record count unexpected",
+				fmt.Sprintf("`%s` has %d rows", w.Date1.Format("2006-01-02"), records),
+				fmt.Sprintf("`%s`, has %d rows", w.Date2.Format("2006-01-02"), records2),
+				fmt.Sprintf("%.3f%% > %.3f%%", math.Abs(p)*100, w.Percent*100),
+			}
+			w.sendSlack(m...)
+			return task.Completed(fmt.Sprintln(m))
+		}
+
+		msg = fmt.Sprintf("variance good at %.2f%%  %s:%d ~ %s:%d",
+			p*100.0, w.Date1.Format("2006-01-02"), records, w.Date2.Format("2006-01-02"), records2)
+	}
+
+	return task.Completed("table %s %s db-check passed for date %s, records: %d %s",
+		w.Table, w.DateField, w.Date1.Format("2006-01-02"), records, msg)
+}
+
+// formats messages and sends the message to slack
+func (w *worker) sendSlack(msg ...string) {
+	log.Println("sending slack alert message")
+	tm := fmt.Sprintf(":heavy_exclamation_mark: *task-tools db-check* `%s`\n", w.Table)
+	for _, m := range msg {
+		tm += ("•         " + m + "\n")
+	}
+	sm := w.slack.NewMessage(tm)
+	err := w.slack.SendMessage(sm)
+	if err != nil {
+		log.Println("error sending slack message", err.Error())
+	}
+}
+
+// date will take a given time.Time and duration and return the time minus that duration
+// truncated to just the date i.e., 2006-01-02
+func date(dt time.Time, duration string) (t time.Time, err error) {
+	if dt.IsZero() {
+		return t, fmt.Errorf("zero time given")
+	}
+
+	d, err := time.ParseDuration(duration)
+	if err != nil {
+		return t, fmt.Errorf("error parsing duration %s %w", duration, err)
+	}
+
+	t = time.Date(dt.Year(), dt.Month(), dt.Day(), 0, 0, 0, 0, time.UTC).Add(-d)
+	return t, nil
+}
+
+func truncDate(date string) (t time.Time) {
+	t, _ = time.Parse("2006-01-02", date)
+	return time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, time.UTC)
+}
```

Comment:
Is this func used?

Reply:
left over from testing various things. I can remove it.

Reply:
If you are using it for testing try `trial.ParseDate`